apps/status-service/src/app/jobs/checkEndpoint.ts

```diff
@@ -280,3 +284,58 @@ const updateAppStatus = async (
 
   return response;
 };
+
+/**
+ * This is an attached job that checks auto change status boolean flag and determines
+ * what status to update based on the health check
+ */
+async function checkAndUpdateAutoChangeStatus(props: CreateCheckEndpointProps) {
+  const { app, serviceStatusRepository, endpointStatusEntryRepository, logger } = props;
+  const QUERY_SIZE = 5;
+
+  try {
+    if (app.autoChangeStatus) {
+      //The history array contains 1 item for each minute.
+      const recentHistory = await endpointStatusEntryRepository.findRecentByUrlAndApplicationId(
+        app.url,
+        app.appKey,
+        QUERY_SIZE
+      );
+
+      //get first and last date time to make sure the time is 5 minutes apart.
+      const mostRecentDate = moment(new Date(recentHistory.at(0).timestamp));
+      const oldestDate = moment(new Date(recentHistory.at(-1).timestamp));
+
+      //Need to round date diff as MomentJS can give you an incorrect number because of the milliseconds
+      const dateDiff = Math.round(mostRecentDate.diff(oldestDate, 'minute'));
+
+      const failedForMoreThanFiveMinutes = recentHistory.filter((hist) => !hist.ok && dateDiff >= 5).length > 0;
```

Comment:
Considering this edge case:
5 mins before: ok,
4 min before: ok,
3 min before: failed,
2 min before: failed,
Last min: failed

This has not been specified in the task. We might need to clarify it later.